packages/datatrak-web/src/features/Survey/Components/SurveyQuestion.tsx

```diff
@@ -70,44 +78,100 @@ export const SurveyQuestion = ({
   type,
   name,
   updateFormDataOnChange,
+  validationCriteria = {},
   ...props
 }: SurveyQuestionFieldProps) => {
-  const { control } = useFormContext();
-  const { setSingleAnswer } = useSurveyForm();
+  const { control, formData, errors } = useFormContext();
+  const { setFormData } = useSurveyForm();
   const FieldComponent = QUESTION_TYPES[type];
 
   if (!FieldComponent) {
     return <QuestionPlaceholder>{name}</QuestionPlaceholder>;
   }
 
+  const controllerName = getNameForController(name, type);
+
   // If the question dictates the visibility of any other questions, we need to update the formData when the value changes, so the visibility of other questions can be updated in real time. This doesn't happen that often, so it shouldn't have too much of a performance impact, and we are only updating the formData for the question that is changing, not the entire formData object.
   const handleOnChange = e => {
     if (updateFormDataOnChange) {
-      setSingleAnswer(name, e.target.value);
+      setFormData({
+        [name]: e.target ? e.target.value : e.value, // autocomplete event returns an object with value, others return an event with a target
+      });
     }
   };
 
+  const { mandatory: required, min, max } = validationCriteria;
+
+  const getRules = () => {
```

Comment:
I think this point is pretty subjective so there no action required but I have a slight preference for defining these logic heavy functions outside of the react component because I find it easier to read. For example this function is just dependant on type but that is not very obvious when it is inline. 

**A couple of other minor pros to pulling them out**
It can also make testing these functions in isolation easy (not that I do that very often yet...).
Some people argue that it's more performant (although I don't give much weight to that)

There are definitely cases where I don't like extracting them out which is when there are lots of arguments (more than 1 or 2) that would need to be passed in. 

Like I said I don't feel too strongly either way but just wanted to mention it in case you agree.